src/main/scala/com/microsoft/hyperspace/index/IndexLogEntry.scala

```diff
@@ -335,16 +352,20 @@ object Hdfs {
   /**
    * Hdfs file properties.
    * @param content Content object representing Hdfs file based data source.
-   * @param appendedFiles Appended files since the last time derived dataset was updated.
-   * @param deletedFiles Deleted files since the last time derived dataset was updated.
+   * @param update Captures any updates since 'content' was created.
    */
-  case class Properties(
-      content: Content,
-      appendedFiles: Option[Content] = None,
-      deletedFiles: Option[Content] = None)
+  case class Properties(content: Content, update: Option[Update] = None)
 }
 
-// IndexLogEntry-specific Relation that represents the source relation.
+/**
+ * IndexLogEntry-specific Relation that represents the source relation.
+ *
+ * @param rootPaths List of root paths for the source relation.
+ * @param data Source data since last time derived dataset was updated.
```

Comment:
This seems to be mentioning the `Update` field?
How about
```
@param data Source data for the relation.
            Hdfs.properties.content captures source data which derived dataset was created from.
            Hdfs.properties.update captures any updates since the derived dataset was created. 
            Note that Hdfs.properties.update.fingerprint is a fingerprint created from the latest source data.
```

Reply:
@sezruby Writing this comment, `Update.fingerprint` still feels out of place. :(

What if we update `SparkPlan.Properties.fingerprint` to always point to the latest fingerprint (instead of having fingerprint inside `Update`)?